service/history/workflow/mutable_state_impl.go

```diff
@@ -2949,7 +3024,9 @@ func (ms *MutableStateImpl) ReplicateUpsertWorkflowSearchAttributesEvent(
 	event *historypb.HistoryEvent,
 ) {
 	upsertSearchAttr := event.GetUpsertWorkflowSearchAttributesEventAttributes().GetSearchAttributes().GetIndexedFields()
+	ms.approximateSize -= ms.executionInfo.Size()
```

Comment:
nit: only compute the size for the search attributes (same for memo)?

Reply:
Those are both maps of `Payload` which doesn't have a Size method to call. I could do the calculation using some other method if you think that would be less expensive than getting the size of the whole executionInfo